astropy/io/fits/hdu/table.py

```diff
@@ -1102,8 +1103,8 @@ def dump(self, datafile=None, cdfile=None, hfile=None, overwrite=False):
                         exist.append(f)
 
         if exist:
-            raise OSError('  '.join([f"File '{f}' already exists."
-                                     for f in exist]))
+            msg1, msg2 = NOT_OVERWRITING_MSG.split('.', maxsplit=1)
+            raise OSError(' '.join([msg1.format(f)+'.' for f in exist]) + msg2)
```

Comment:
This goes with my suggestion in the template wording.
```suggestion
            raise OSError(NOT_OVERWRITING_MSG.format(', '.join(exist)))
```

Reply:
As far as I can tell this is the only place where the error can be raised for multiple files simultaneously. Would be it better to have a different error message here?

Reply:
Well, this is FITS after all. I don't think anyone will complain if you omit this file from the patch altogether... 😆